### PR TITLE
Adding bias argument to NN normalization methods

### PR DESCRIPTION
**Pull request by ignasa007**
Fixes #157144

First commit - updated Batch/Instance Norm implementations to match LayerNorm's, in the sense that a `bias: bool` argument controls whether bias is used (only relevant if `affine: bool = True`). Default value for `bias` is equal to that of `affine` &ndash; `True` for BatchNorm and `False` for InstanceNorm. 

C/C++ implementations accommodate `weight is not None` but `bias is None`, so changing the Python API is sufficient. I tried to find all instances in the repository where Batch/Instance Norms were being initialized using positional args, where trouble could arise since `bias` is positioned between `affine` and `track_running_stats`. I have added the same value for `bias` as the one used for `affine` in these cases.

### CODE REVIEW

**Comment by ignasa007** [comment]
I have left RMSNorm for now because I can't navigate the C++ files to make all the necessary changes for adding the bias parameter.

**Comment by ignasa007** [comment]
Fixed regression errors:
1. InstanceNorm uses default `affine=False`, and I had accordingly set default `bias=False`. But in certain tests, `affine=True` is passed, in which case it was expected for `bias` to be `True`, but since these initializations doesn't set the `bias` parameter explicitly, it takes default value `False`, resulting in errors down the road.
2. I had accidentally set `bias : bool` as a constant in GroupNorm, which was resulting in MyPy errors.

**Comment by ignasa007** [comment]
Hi @jbschlosser and @mikaylagawarecki, can I check if this is a desired feature, and if so, does the change look good? If not, I'll close the PR.

**Comment by mikaylagawarecki** [comment]
Hey @ignasa007 this feature sounds reasonable, sorry for the delay on getting back to you. Let me review the PR later today

**Comment by ignasa007** [comment]
Addressed linter warnings. Also added `bias` argument in `SyncBatchNorm`'s `super().__init__` call.

File: torch/nn/modules/batchnorm.py [hunk 1]
```diff
@@ -41,6 +41,7 @@ def __init__(
         eps: float = 1e-5,
         momentum: Optional[float] = 0.1,
         affine: bool = True,
+        bias: bool = True,
```
Comment (mikaylagawarecki) [inline torch/nn/modules/batchnorm.py]: Unfortunately, adding `bias` here like this would break backward compatibility (and similarly for everything below) because the meaning of a call like `Mod(5, 1e-5, 0.1, True, False)` would change

Let's move bias to the end of the argument list and also make it kwarg-only, i.e. `__init__(..., *, bias: bool =True)`

File: torch/nn/modules/batchnorm.py [hunk 2]
```diff
@@ -227,13 +240,15 @@ def __init__(
             momentum,
             False,
             False,
+            False,
```
Comment (mikaylagawarecki) [inline torch/nn/modules/batchnorm.py]: why is this `False`? if the reason is similar to that for affine and track_running_stats could you update the comment please

File: torch/ao/nn/intrinsic/qat/modules/linear_fused.py [hunk 3]
```diff
@@ -50,7 +50,7 @@ def __init__(
         assert qconfig, "qconfig must be provided for QAT module"
         self.qconfig = qconfig
         self.freeze_bn = freeze_bn if self.training else True
-        self.bn = nn.BatchNorm1d(out_features, eps, momentum, True, True)
+        self.bn = nn.BatchNorm1d(out_features, eps, momentum, True, True, True)
```
Comment (mikaylagawarecki) [inline torch/ao/nn/intrinsic/qat/modules/linear_fused.py]: Let's make `bias` kw-only (see below comment re backward compatibilty)

**Review by mikaylagawarecki — COMMENTED** [review]
Thank you for your patience and for sending this change, see the comment re backward compatibility 

Do also add `ModuleInputs` with the `bias` argument to the relevant module_inputs functions to test that the newly added arguments work, see the existing example
https://github.com/pytorch/pytorch/blob/ee4c5c7cd28644c0c7181ea09d0d6ade295c5a6b/torch/testing/_internal/common_modules.py#L1928-L1931

File: torch/nn/modules/batchnorm.py (hunk 1 above)
Comment (ignasa007) [reply torch/nn/modules/batchnorm.py]: That makes a lot more sense than what I was doing!

File: torch/nn/modules/batchnorm.py (hunk 2 above)
Comment (ignasa007) [reply torch/nn/modules/batchnorm.py]: Updated!

File: torch/ao/nn/intrinsic/qat/modules/linear_fused.py (hunk 3 above)
Comment (ignasa007) [reply torch/ao/nn/intrinsic/qat/modules/linear_fused.py]: Done!

**Comment by ignasa007** [comment]
I'm really sorry &ndash; not sure where the GroupNorm errors are coming from, and I can only diagnose in a few days now (working on NeurIPS rebuttal). I'll try to come back with a good update, soon. Sorry, again, for taking so long to wrap this up.

**Comment by mikaylagawarecki** [comment]
reopening

**Comment by mikaylagawarecki** [comment]
@ignasa007 I'm sorry, could you open a new PR please, for some reason I cannot reopen this one

**Comment by ignasa007** [comment]
No worries, reopened at [https://github.com/pytorch/pytorch/pull/176573](https://github.com/pytorch/pytorch/pull/176573).
